### DISCUSSION TITLE
Flux Select for selecting a model

### DISCUSSION THREAD

**Discussion by mariomeyer** [livewire/flux#732] [Feature Requests]
The idea is to make Flux select be able to iterate through Collections and **magically** select one of them just like MaryUI does (https://mary-ui.com/docs/components/select)

Let's say I have a blade component for a `$product` (App\Models\Product) but that product has `$product->variations` (App\Models\Product\Variation).

It would be lovely to not have to manually generate the iteration on the blade and create methods for setting the selected variation based on the ID. This could all be wrapped by `flux:select`.

Something like:

```php
@php
    Product $product = App\Models\Product::find(1);
    Variation $variation = $product->variations->first();
@endphp
 
<flux:select label="Select Variation" icon="rectangle-stack" :options="$product->variations" wire:model="variation" />
```

**Comment by jeffchown** [comment]
@mariomeyer To me, that's a specialized use of `flux:select` that should be created as custom component that wraps the `flux:select` as there can be many different configurations of the :options objects/arrays.

**Reply by ju5t** [reply]
This is what we do. We have our own logic inside that wrapper component to normalize data.
